### PR TITLE
Fix importing of function's signature from IDA

### LINKED ISSUE #1492: IDA plugin don't export prototype of function
**Describe the bug**
IDA exporter don't export function's prototypes correctly: `ida_name.get_demangled_name` and `idaapi.print_type` often returns only name of function without parameters (see [code](https://github.com/NationalSecurityAgency/ghidra/blob/dc16f297cc3b55816f929f25f38b5cc779631bce/GhidraBuild/IDAPro/Python/7xx/python/idaxml.py#L967-L980)).

**To Reproduce**
1. Open IDA database or analyze new file.
2. Create function with parameters.
3. Export IDA database to XML.
4. Import XML to Ghidra (see Fig. 1, that shows the problem).

**Expected behavior**
Information about our function should be imported with all parameters and datatypes, that are used for parameters.

**Screenshots**

> Fig. 1. List of functions, that were not imported

**Attachments**
You can use any file for testing.

**Environment (please complete the following information):**
 - OS: Linux 5.4.3
 - Java Version: 11.0.5
 - Ghidra Version: dc16f297cc3b55816f929f25f38b5cc779631bce

**Additional context**
I create PR for that issue. I also had to make some changes to Ghidra's code, that could affect issues as #651, #857.

Also I found, that Ghidra don't parse calling convention sadly (besides `cdecl`). It's mean, that exporting/importing from Ghidra to Ghidra don't works correctly (what of course is warned).

### PR DESCRIPTION
**Pull request by saruman9**
cfce5fa fix problem of exporting thunked function:
```
ERROR (FunctionsXmlMgr) Failed to create function at 00004294: Unable to create function at 00004294
```
The commit may be regressive, but I used Ghidra with merged the PR for a month and there were no problems.

Sadly Ghidra don't support parsing of many calling convention (see `TODO` in the code). I don't change CParser, because it will be very massive for the PR.

Fixes #1492.

### CODE REVIEW

**Comment by saruman9** [comment]
Any review?

**Comment by saruman9** [comment]
Any updates?

File: Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java [hunk 1]
```diff
@@ -281,7 +280,8 @@ private boolean createFunction(TaskMonitor monitor, String funcName,
 				if (!recreateFunction) {
 					long bodySize = functionContaining.getBody().getNumAddresses();
 					if (bodySize != 1) {
-						return false;
+						newFunc = functionContaining;
```
Comment (ryanmkurtz) [inline Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java:283]: Can you explain why changes to `CreateFunctionCmd.java` are needed?

**Comment by ghidra1** [comment]
Our PROGRAM.DTD needs a big update to catchup with our API and is rather constrained.  If you will work with us we can probably adapt it better to our API.  We have always worried about exceeding the capabilities of IDA for function related details.  We could probably start a discussion where we could propose PROGRAM.DTD changes if someone like yourself could adapt the IDA import/export plugins.

**Comment by saruman9** [comment]
Sounds great! Unfortunately, I no longer have a license for IDA to help you. If I have it, then I will continue working in this direction. BTW I think that a Discussion will be useful, because many people use the plugin to transfer information to/from IDA.

File: Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java (hunk 1 above)
Comment (saruman9) [reply Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java:283]: Unfortunately, due to the fact that the changes were made more than three years ago, I do not remember the details in this regard. Everything I know is written in the description of the comit. Unfortunately, now I no longer have a license for IDA to reproduce the behavior of the plugin to find out why I made such changes. Please excuse me for not being able to explain the changes made.

**Comment by ryanmkurtz** [comment]
Closing this due to it getting very out of date, and no one having IDA to properly test it.

**Pull request closed without merging**